### DISCUSSION TITLE
Error handling in multistep form

### DISCUSSION THREAD

**Discussion by glRajkumar2** [react-hook-form/react-hook-form#10447] [Q&A]
following is my `Main.js` file
```js
  const [step, setStep] = useState(1)
  const methods = useForm()

  <FormProvider {...methods} >
    <form onSubmit={methods.handleSubmit(onSubmit)}>
      {step === 1 && <Step1 setStep={setStep} />}
      {step === 2 && <Step2 setStep={setStep} />}
      {step === 3 && <Step3 setStep={setStep} />}
    </form>
  </FormProvider>
```
using `useFormContext` to hadle data in `Step1`, `Step2`,`Step3`.
in `Step3` there is submit button. all fields are required. When I am trying to submit form without adding any data, the form is submiting. errors not cached by react-hook-form. (I added required field in all places)

**Comment by Moshyfawn** [comment]
It's not enough info to go off of 😉 Please, consider creating a reproduction codesandbox instead

**Reply by glRajkumar2** [reply]
here is the code sand box [link](https://codesandbox.io/s/agitated-hooks-s23sy6?file=/src/Main.js)

**Comment by Moshyfawn** [comment]
Seems like what you're experiencing is related to #10043

> unmounted input will not be validated

I'd advice you to try and use [validation schema](https://react-hook-form.com/docs/useform#resolver) instead.
